libvast/vast/bloom_filter_synopsis.hpp

```diff
@@ -0,0 +1,89 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/bloom_filter.hpp"
+
+#include <caf/deserializer.hpp>
+#include <caf/optional.hpp>
+#include <caf/serializer.hpp>
+
+#include <vast/detail/assert.hpp>
+#include <vast/synopsis.hpp>
+#include <vast/type.hpp>
+
+namespace vast {
+
+/// A Bloom filter synopsis.
+template <class T, class HashFunction>
+class bloom_filter_synopsis : public synopsis {
+public:
+  using bloom_filter_type = bloom_filter<HashFunction>;
+  using hasher_type = typename bloom_filter_type::hasher_type;
+
+  bloom_filter_synopsis(vast::type x, bloom_filter_type bf)
+    : synopsis{std::move(x)}, bloom_filter_{std::move(bf)} {
+    // nop
+  }
+
+  void add(data_view x) override {
+    bloom_filter_.add(caf::get<view<T>>(x));
+  }
+
+  caf::optional<bool>
+  lookup(relational_operator op, data_view rhs) const override {
+    switch (op) {
+      default:
+        return caf::none;
+      case equal:
+        return bloom_filter_.lookup(caf::get<view<T>>(rhs));
+      case in: {
+        if (auto xs = caf::get_if<view<set>>(&rhs)) {
+          for (auto x : **xs)
```

Comment:
```suggestion
          VAST_ASSERT(*xs);
          for (auto x : **xs)
```

Reply:
This is not a pointer, actually. The handle has pointer semantics, see `view.hpp`. So this actually doesn't even work.

Reply:
Smells like bad API design to me. Is there some member function we can use over `operator*` to make this clearer?